pkg/scheduler/dependency_graph.go

```diff
@@ -381,11 +426,8 @@ func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.D
 	}
 
 	sort.Stable(&tmp)
-	var result []string
-	for _, replica := range tmp.Items {
-		result = append(result, replica.ReplicaName())
-	}
-	return result[:count], nil
+
+	return tmp.Items[:count], nil, nil
 }
 
 // BuildDependencyGraph loads dependencies data and creates the DependencyGraph
```

Comment:
2 lines you pointed to belongs to different functions. How it can be extracted? Which one of the 2 functions grown too big?